File: backend/app/schemas/admin_contact.py
import uuid
from datetime import datetime

from pydantic import BaseModel, EmailStr, Field


class ContactSubmissionCreate(BaseModel):
    """Public /contact page form payload — matches ContactForm.tsx's fields."""

    name: str = Field(min_length=1, max_length=200)
    email: EmailStr
    organization: str | None = Field(default=None, max_length=200)
    subject: str = Field(min_length=1, max_length=100)
    message: str = Field(min_length=20, max_length=2000)


class ContactReplyCreate(BaseModel):
    reply_message: str = Field(min_length=1)


class ContactSubmissionAdminSummary(BaseModel):
    id: uuid.UUID
    name: str
    email: str
    organization: str | None
    subject: str
    status: str
    created_at: datetime

    model_config = {"from_attributes": True}


class ContactSubmissionAdminDetail(BaseModel):
    id: uuid.UUID
    name: str
    email: str
    organization: str | None
    subject: str
    message: str
    status: str
    reply_message: str | None
    replied_by_name: str | None
    replied_at: datetime | None
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/lib/types/admin-contact.ts
// Mirrors backend/app/schemas/admin_contact.py exactly — keep these two in sync.

export type ContactSubmissionStatus = "new" | "replied";

export interface ContactReplyCreate {
  reply_message: string;
}

export interface ContactSubmissionAdminSummary {
  id: string;
  name: string;
  email: string;
  organization: string | null;
  subject: string;
  status: ContactSubmissionStatus;
  created_at: string;
}

export interface ContactSubmissionAdminDetail {
  id: string;
  name: string;
  email: string;
  organization: string | null;
  subject: string;
  message: string;
  status: ContactSubmissionStatus;
  reply_message: string | null;
  replied_by_name: string | null;
  replied_at: string | null;
  created_at: string;
}

// Mirrors ContactForm.tsx's SUBJECTS keys — the submitted `subject` field is
// the internal value (e.g. "data_request"), not the display label; this maps
// it back for admin display.
export const CONTACT_SUBJECT_LABELS: Record<string, string> = {
  data_request: "Dataset Request Help",
  technical: "Technical Support",
  partnership: "Partnership & Collaboration",
  data_quality: "Data Quality Issue",
  api: "API & Integration",
  media: "Media & Press",
  general: "General Enquiry",
};
